### PR TITLE
CTED-2369 post to groups, pass correct groupid in create object

### PR DESCRIPTION
**Pull request by loyisongq**

### CODE REVIEW

File: classes/controller/edit_controller.php [hunk 1]
```diff
@@ -212,27 +212,35 @@ public function update_post_action() {
             $post       = $DB->get_record('hsuforum_posts', array('id' => $postid), '*', MUST_EXIST);
             $discussion = $DB->get_record('hsuforum_discussions', array('id' => $post->discussion, 'forum' => $forum->id), '*', MUST_EXIST);
 
-            if (empty($groupid)) {
-                $groupid = -1;
+            foreach ($groupids as $gid) {
+                if (empty($groupids)) {
+                    $groupids[] = -1;
+                } else {
```
Comment (BrendonPretorius) [inline classes/controller/edit_controller.php:218]: If it is empty, the foreach won't iterate and this code won't execute. Please take another look at this

**Review by BrendonPretorius — CHANGES_REQUESTED** [review]

File: classes/controller/edit_controller.php [hunk 2]
```diff
@@ -212,27 +212,29 @@ public function update_post_action() {
...
+                'groupids'      => $groupids,
                 'itemid'        => $itemid,
                 'message'       => $message,
                 'messageformat' => $messageformat,
                 'reveal'        => $reveal,
                 'privatereply'  => $privatereply,
                 'timestart'     => $timestart,
                 'timeend'       => $timeend
-            ));
+                ));
```
Comment (BrendonPretorius) [inline classes/controller/edit_controller.php:237]: Is this indentation correct?

**Review by BrendonPretorius — CHANGES_REQUESTED** [review]
Please share screenshot of moodle code checker run on files you've changed. Please only fix what has been highlighted for lines that you have changed

File: classes/controller/edit_controller.php (hunk 2 above)
Comment (loyisongq) [reply classes/controller/edit_controller.php]: Screen Shot 2022-04-01 at 1 57 02 PM
Screen Shot 2022-04-01 at 1 54 54 PM
Screen Shot 2022-04-01 at 1 44 46 PM

**Review by BrendonPretorius — APPROVED** [review]

**Pull request merged**